### PR TITLE
Remove deprecated `binned` metadata

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Remove deprecated `binned` metadata,
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] update tests,
- [x] ready for review.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -22,6 +22,9 @@ migration of the IO plugins to `RosettaSciIO
 - :ref:`NeXus data format <nexus-format>`: The `dataset_keys`, `dataset_paths`
   and `metadata_keys` keywords were removed. Use `dataset_key`, `dataset_path`
   and `metadata_key` instead.
+- ``metadata.Signal.binned`` is removed, use the ``is_binned`` axis attribute
```
Comment (jlaehne) [inline CHANGES.rst:141]: This is the rsciio-split related API removal section.

When merging in Release next major, you can amend it to the other API removals.
(the duplicate `API Removal` headline has to be then removed as well)

File: CHANGES.rst [hunk 2]
```diff
@@ -137,6 +137,12 @@ Data Visualization
 - The keyword argument ``line_style`` of :py:func:`~.drawing.utils.plot_spectra` have been renamed to ``linestyle``.
 
 
+Signal
```
Comment (jlaehne) [inline CHANGES.rst:139]: The different subsections of API Removal are arbitrarily ordered. Should we order them alphabetically? Or move the two large chunks Machine Learning and Model fitting after the more concise one?

**Review by jlaehne — APPROVED** [review]

File: CHANGES.rst (hunk 2 above)
Comment (jlaehne) [reply CHANGES.rst:139]: I added this as a commit, as it improves readability of the changelogs.

Comment (ericpre) [reply CHANGES.rst:139]: Yes, and I would leave it to the review before the release - I added it to the task list of https://github.com/hyperspy/hyperspy/issues/2996#issue-1350647756.

**Pull request merged**